ArduCopter/GCS_Mavlink.cpp

```diff
@@ -134,28 +134,22 @@ NOINLINE void Copter::send_extended_status1(mavlink_channel_t chan)
 
 void NOINLINE Copter::send_location(mavlink_channel_t chan)
 {
-    uint32_t fix_time;
-    // if we have a GPS fix, take the time as the last fix time. That
-    // allows us to correctly calculate velocities and extrapolate
-    // positions.
-    // If we don't have a GPS fix then we are dead reckoning, and will
-    // use the current boot time as the fix time.
-    if (gps.status() >= AP_GPS::GPS_OK_FIX_2D) {
-        fix_time = gps.last_fix_time_ms();
-    } else {
-        fix_time = millis();
+    const uint32_t now = AP_HAL::millis();
```

Comment:
Can we ask the EKF when it calculated the position and use that here instead ? That would inprove the jitter that I was talking about in the dev meeting.

Reply:
@amilcarlucas, before I made this change to use system time the position reported could be up to 200ms off but after this change it should be down to a max of 2.5ms.  I suspect the transmission times of most radios will drown out this difference... no?

Reply:
Yes... but for high speed, RTK systems those 2.5 ms might get interesting. How hard is to get them ?

Reply:
@amilcarlucas, I'm sure it's possible but I'm not sure how difficult it would be.  We still have a ways to go before the this level of lag the biggest issue i think.